tests/template.go

```diff
@@ -0,0 +1,82 @@
+package tests
+
+import (
+	"log"
+	"os"
+	"os/exec"
+
+	. "github.com/onsi/ginkgo"
+)
+
+func BuildGinTemplate() {
+	shCmd := exec.Command("go", "build", "-o", os.Getenv("GOPATH")+"/bin/nimble-gin", "github.com/nimblehq/gin-templates")
+	shCmd.Stdout = os.Stdout
+	err := shCmd.Run()
+	if err != nil {
+		Fail("Failed to build gin-templates: " + err.Error())
+	} else {
+		log.Println("Gin-templates built successfully.")
+	}
+}
+
+func CreateProjectFromGinTemplate() string {
+	input := []byte("test-gin-templates")
+
+	reader, writer, err := os.Pipe()
+	if err != nil {
+		Fail("Failed to create file to store stdin value: " + err.Error())
+	}
+
+	_, err = writer.Write(input)
+	if err != nil {
+		Fail("Failed to write stdin value to file: " + err.Error())
+	}
+	writer.Close()
+
+	var shCmd *exec.Cmd
+	if os.Getenv("BRANCH") == "" {
+		shCmd = exec.Command("nimble-gin", "create")
+	} else {
+		shCmd = exec.Command("nimble-gin", "create", "-b", os.Getenv("BRANCH"))
+	}
+
+	shCmd.Stdout = os.Stdout
+	shCmd.Stdin = reader
+
+	err = shCmd.Run()
+	if err != nil {
+		Fail("Failed to create template: " + err.Error())
+	} else {
+		log.Println("Template created successfully.")
+	}
+
+	currentPath, err := os.Getwd()
+	if err != nil {
+		Fail("Failed to get current directory: " + err.Error())
+	}
+
+	return currentPath
+}
+
+func DownloadGinTemplate() {
```

Comment:
Just minor preference to order the methods of this file ordered by the use case:

```go
func DownloadGinTemplate() {}
func BuildGinTemplate() {}
func RemoveCookiecuttersCache() {}
func CreateProjectFromGinTemplate() string {}
```

As a general rule, prefer ordering methods how they are used logically, not how/when the methods were added. It's better for code discoverability, readability and maintenance.

Reply:
Fixed in 146811705e93b2b23a0b069ce136ee88f740c1e2